measurement/plugins/netflix_fast/measurements.py

```diff
@@ -0,0 +1,334 @@
+
+All these results are then returned as a list.
+"""
+
+import requests
+import re
+import time
+import urllib
+import json
+from threading import Thread
+from collections import deque
+from statistics import mean
+
+from measurement.measurements import BaseMeasurement
+from measurement.results import Error
+from measurement.units import RatioUnit, TimeUnit, StorageUnit, NetworkUnit
+from measurement.plugins.latency.measurements import LatencyMeasurement
+from measurement.plugins.download_speed.measurements import DownloadSpeedMeasurement
+from measurement.plugins.netflix_fast.results import (
+    NetflixFastMeasurementResult,
+    NetflixFastThreadResult,
+)
+
+
+NETFLIX_ERRORS = {
+    "netflix-err": "Netflix test encountered an unknown error",
+    "netflix-ping": "Netflix test encountered an error when pinging hosts",
+    "netflix-response": "Netflix test received an invalid response from fast.com",
+    "netflix-script-regex": "Netflix test failed to find script in the response",
+    "netflix-script-response": "Netflix test received an invalid response from script",
+    "netflix-token-regex": "Netflix test failed to find token in the response",
+    "netflix-api-response": "Netflix test received an invalid response when querying for URLs",
+    "netflix-api-json": "Netflix test failed to decode URLs",
+    "netflix-api-parse": "Netflix test failed interpret elements of the decoded JSON",
+    "netflix-connection": "Netflix test failed to connect to download URLs",
+    "netflix-download": "Netflix test encountered an error downloading data",
+}
+CHUNK_SIZE = 64 * 2 ** 10
+MIN_TIME = 3
+PING_COUNT = 4
+STABLE_MEASUREMENTS_LENGTH = 6
+STABLE_MEASUREMENTS_DELTA = 2
+
+
+class NetflixFastTestMeasurement(BaseMeasurement):
+    def __init__(
+        self,
+        id,
+        urlcount=3,
+        max_time=30,
+        sleep_seconds=0.2,
+        terminate_on_thread_complete=True,
+        terminate_on_result_stable=True,
+    ):
+        super(NetflixFastTestMeasurement, self).__init__(id=id)
+        self.id = id
+        self.urlcount = urlcount
+        self.max_time = max_time
+        self.sleep_seconds = sleep_seconds
+        self.terminate_on_thread_complete = terminate_on_thread_complete
+        self.terminate_on_result_stable = terminate_on_result_stable
+        self.finished_threads = 0
+        self.exit_threads = False
+        self.total = 0
+        self.sessions = []
+        self.client_data = {"asn": None, "ip": None, "isp": None, "location": None}
+        self.targets = []
+        self.thread_results = [
+            {
+                "index": i,
+                "elapsed_time": None,
+                "download_size": 0,
+                "download_rate": 0,
+                "url": None,
+                "location": None,
+            }
+            for i in range(self.urlcount)
+        ]
```

Comment:
This could be considered a bit misleading that there will be results here on initialization but nothing actually run.